components/root/root.jsx

```diff
@@ -266,7 +266,7 @@
             landing = desktopAppDownloadLink;
         }
 
-        if (landing && !BrowserStore.hasSeenLandingPage() && !toResetPasswordScreen && !this.props.location.pathname.includes('/landing') && !this.props.location.href.includes('.test.mattermost.com') && !UserAgent.isDesktopApp()) {
+        if (landing && !BrowserStore.hasSeenLandingPage() && !toResetPasswordScreen && !this.props.location.pathname.includes('/landing') && !this.props.location.href?.includes('.test.mattermost.com') && !UserAgent.isDesktopApp()) {
```

Comment:
@pfltdv I hope you can answer this, Do I need to do anything here ?

Reply:
^ CCing @nevyangelova since she was involved in that change as well. Instead of checking the whole URL, could we check if `location.origin.endsWith('.test.mattermost.com')` to fix the issue while still satisfying the CodeQL check?

Reply:
Code scanning will always complain about string operations on the url and recommends to use map or array to identify allowed domains. https://github.com/mattermost/mattermost-webapp/security/code-scanning/54

Since we are using wildcard domains maybe below regex may help
```
this.props.location.hostname.search(/^.*\.test\.mattermost\.com$/g)==0
```